Sources/Commands/PackageCommands/Describe.swift

```diff
@@ -41,12 +41,31 @@ extension SwiftPackageCommand {
                 observabilityScope: swiftCommandState.observabilityScope
             )
 
-            try self.describe(package, in: type)
+            let buildSystem = try await swiftCommandState.createBuildSystem(
```

Comment:
WDYT about making it conditional on `// swift-tools-version: 6.1`? I would expect some CLI relies on the old behavior. This change seems additive, but I'd rather see it only applied for 6.1 package to make this at least somewhat opt-in.

Reply:
It feels a bit over cautious to me. Perhaps we have some very tight couplings out there that we need to consider?

For the text format, I would be surprised if another tool directly consumes that. I feel that it's mostly to help end-users understand what's contained in their package.

A tool that consumes the output would most likely use the JSON format. Most JSON decoders that I know are capable of ignoring additional fields and objects.

Reply:
I'm wondering if this should be a separate subcommand. Partly to avoid surprises, but also I'd like to see this used by bash completions and friends and I don't think we want those things parsing the output of describe. For that matter, I'm not sure we want users to be doing that either :).

Reply:
What's the concern around completion processors using describe to find targets, products, etc. for completing various swift commands? I thought that this might be one of the intended use cases since describe has the JSON output format option. Maybe we're aiming towards a new direction for tools to query package metadata?

Reply:
Oh, wait, we have a ```completion-tool``` subcommand! Can you check whether this supports what you're trying to accomplish?

Also, looking at describe right now, It's help does say ```OVERVIEW: Describe the current package```. By listing all executables you are changing what describe does, no? It's is only listing things in the root package. That's why listing more than that is really a different subcommand. And it will cause package resolution to happen which will take a long time. ```describe``` does not do that.

Reply:
Thanks @dschaefer2 I'll check those out.

My rationale for adding the transitive executables to describe was that the package enables you to run all of those executables in the same way you can run the ones in the package. They form part of the package's interface description. Projects intend for their devs to run some of those tools, such as formatters. Plus, people already know about the describe subcommand, which can help with discoverability.

Ah, so it's package resolution that's the concern here. Is it because it can be slow, or something else? Is it much slower than package analysis, which I believe requires compiling and running the Package.swift. Other subcommands like `swift run` do this, but then I suppose that there is an expectation that analysis, resolution, build will happen before the executable is run.

Reply:
describe compiles and runs the Package.swift. That's fine. In fact, that's what it's doing, showing the results of that. The resolve does all the checkout and deep dependency analysis. That, of course, takes a while.

It's also the sheer volume of stuff coming out in describe when all I want to see what executables I can run. I just ran describe on swiftpm itself, a pretty standard sized package. Pages and pages of stuff. It's a debug tool, not really intended for human consumption. I don't think I'd want to be using it as a general query mechanism.

Reply:
Thanks all, I've reworked this patch so that it instead adds a `swift package show-executables` subcommand modelled after the existing `swift package show-dependencies` subcommand, which is expected to not only analyze the current package, but also resolve packages transitively to find the entire list of products. I think that this should help to decrease the risk of disruption to existing clients of the describe function.